discovery-provider/src/api/v1/tracks.py

```diff
@@ -129,40 +132,85 @@ class Track(Resource):
     @record_metrics
     @ns.doc(
         id="""Get Track""",
+        description="""Gets a track by its encoded ID""",
         params={"track_id": "A Track ID"},
         responses={200: "Success", 400: "Bad request", 500: "Server error"},
     )
     @ns.marshal_with(track_response)
     @cache(ttl_sec=5)
     def get(self, track_id):
-        """Fetch a track."""
         decoded_id = decode_with_abort(track_id, ns)
         return get_single_track(decoded_id, None, ns)
 
 
-track_slug_parser = reqparse.RequestParser()
-track_slug_parser.add_argument("handle", required=False)
-track_slug_parser.add_argument("slug", required=False)
-track_slug_parser.add_argument("route", action="append", required=False)
+full_track_parser = current_user_parser.copy()
+full_track_parser.add_argument("handle")
+full_track_parser.add_argument("url_title")
+full_track_parser.add_argument("show_unlisted", type=inputs.boolean)
+
+
+@full_ns.route(TRACK_ROUTE)
+class FullTrack(Resource):
+    @record_metrics
+    @full_ns.doc(
+        id="""Get Track""",
+        description="""Gets a track by its encoded ID. If `show_unlisted` is true, then `handle` and `url_title` are required.""",
+        params={
+            "track_id": "The ID of the track",
+            "handle": "The User handle of the track owner",
+            "url_title": "The URLized title of the track",
+            "show_unlisted": "Whether or not to show unlisted tracks",
+        },
```

Comment:
Only track_id in url param, rest can be moved to reqparser since there are query params